### PR TITLE
build(snap): update snap package and add snap section in README

### PR DESCRIPTION
**Pull request by MonicaisHer**
**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rest-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

```
git clone https://github.com/MonicaisHer/device-rest-go.git
cd device-rest-go
git checkout update-snap-package 
snapcraft --shell 

snap try prime
snap start --enable edgex-device-rest
snap install edgexfoundry --edge
snap connect edgexfoundry:edgex-secretstore-token edgex-device-rest:edgex-secretstore-token
sudo journalctl -f | grep rest
```
validated by seeing:
```
msg="Starting device-rest 2.0.1-dev.14 "
msg="device rest started"
```

### CODE REVIEW

**Comment by MonicaisHer** [comment]
Could you please review it? Thanks! @siggiskulason @farshidtz

File: README.md [hunk 1]
```diff
@@ -141,4 +141,11 @@ Since `Commanding` is not implemented, specifying `AutoEvents` in the configurat
...
 
         make build
+### Running the device-rest Service as a Snap
+EdgeX Device REST is also available as a snap package. Install the snap with the following command:
+```
+sudo snap install edgex-device-rest
+```
+for more details on the Device REST Snap, including installation, configuration, please refer to [EdgeX REST Device Service Snap](https://github.com/edgexfoundry/device-rest-go/tree/main/snap)
 
+for more details on Snap, including EdgeX Snap, viewing logs, security services, please check [Getting Started with Snap](https://docs.edgexfoundry.org/2.0/getting-started/Ch-GettingStartedSnapUsers/)
```
Comment (farshidtz) [inline README.md:151]: 2x ~for~ For 

more details on ~Snap, including EdgeX Snap~ deploying EdgeX with Snaps, viewing logs

File: snap/local/hooks/go.mod [hunk 2]
```diff
@@ -1,5 +1,6 @@
 module github.com/edgexfoundry/device-rest-go/hooks
 
-require github.com/canonical/edgex-snap-hooks/v2 v2.0.3
+require github.com/canonical/edgex-snap-hooks/v2 v2.1.1
```
Comment (farshidtz) [inline snap/local/hooks/go.mod:3]: Please tidy and check-in the `go.sum` as well.

File: README.md (hunk 1 above)
Comment (MonicaisHer) [reply README.md]: Updated. Thanks!

File: snap/local/hooks/go.mod (hunk 2 above)
Comment (MonicaisHer) [reply snap/local/hooks/go.mod:3]: Added. Thanks!

**Comment by farshidtz** [comment]
Thanks @MonicaisHer.
Could you also add the following to you PR:
1. Change track to 2.1 in snap/README.md
2. Add the following configs:
```
[SecretStore]
secret-store.secrets-file               // SecretStore.SecretsFile
secret-store.disable-scrub-secrets-file // SecretStore.DisableScrubSecretsFile
```

**Review by farshidtz — APPROVED** [review]

**Review by farshidtz — APPROVED** [review]
Looks good. Did some smoke testing to verify that changes to snapcraft.yml and new config items work.

@siggiskulason please squash and merge.

**Comment by MonicaisHer** [comment]
@farshidtz Thanks for your review. I also updated base core from 18 to 20 at the same time as your lastest comment (sorry for the inconvenience). I tested core20 with edgex-cli, and worked as well as core18.

**Review by farshidtz — APPROVED** [review]

**Review by siggiskulason — APPROVED** [review]
LGTM

**Pull request merged**